.github/workflows/upload-algolia.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations  # noqa: INP001
+
+import json
+import os
+from urllib.request import urlopen
+
+from algoliasearch.search_client import SearchClient
+
+api_key = os.getenv("ALGOLIA_API_KEY")
+app_id = os.getenv("ALGOLIA_APP_ID")
+index_name = "test_IBIS"  # os.getenv("ALGOLIA_INDEX")
+index_file = os.getenv("QUARTO_INDEX_PATH")  # search.json
+
+
+def truncate_string(string, max_size):
+    # Encode the string to bytes using UTF-8 encoding
+    encoded_string = string.encode("utf-8")
+
+    # Truncate the bytes to ensure the size is smaller than max_bytes
+    truncated_bytes = encoded_string[: max_size - 1]
+
+    # Decode the truncated bytes back to string
+    truncated_string = truncated_bytes.decode("utf-8", errors="ignore")
+
+    return truncated_string
+
+
+def main():
+    # Download the index generated by quarto from the ibis website
+    with urlopen("https://ibis-project.org/search.json") as response:
+        with open(index_file, "wb") as f:
+            f.write(response.read())
```

Comment:
Do we actually need to write to disk here? How about calling `json.load(f)` directly instead of writing and then reading?

Reply:
Actually, how about `json.loads(response.read().decode("utf-8"))`?